src/Microsoft.DotNet.Cli.Utils/CommandResolution/DefaultCommandResolver.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.ProjectModel.Graph;
+using Microsoft.Extensions.PlatformAbstractions;
+using NuGet.Frameworks;
+using NuGet.Packaging;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    public class DefaultCommandResolver : CompositeCommandResolver
```

Comment:
This is effectively the Policy right? Why not have a DefaultCommandResolverPolicy that returns a CompositeCommandResolver with the right resolvers inside.

I mean, this seems fine to me, but when we chat about it, you keep talking about policies, which indicates that the real abstraction you see here. So, give in to your brain. :D